cmd/cluster/openstack/create_test.go

```diff
@@ -130,3 +131,178 @@ func TestCreateCluster(t *testing.T) {
 		})
 	}
 }
+
+func TestExtractCloud(t *testing.T) {
+	dumpYAML := func(t *testing.T, path string, contents map[string]any) {
+		data, err := yaml.Marshal(contents)
+		if err != nil {
+			t.Fatalf("failed to marshal YAML: %v", err)
```

Comment:
It's preferred to use `t.Errorf` instead of `t.Fatalf` in order to avoid the test termination before see all the results.

Reply:
I think `t.Fatalf` is expected here: an error here would indicate that we failed to marshall/write our test data to disk. If that happens, continuing would be misleading, since it could suggest an issue in the code-under-test rather than the test environment.

I used `t.Error` where appropriate in the test cases since in those cases I do want to see all results, as you note.